gateway-server/src/main/java/org/apache/knox/gateway/services/token/impl/DefaultTokenStateService.java

```diff
@@ -43,9 +48,19 @@
 
   private final Map<String, Long> maxTokenLifetimes = new HashMap<>();
 
+  /* token eviction interval in seconds, default is 5 mins */
+  private long tokenEvictionInterval;
+
+  private final ScheduledExecutorService evictionScheduler = Executors.newScheduledThreadPool(1);
```

Comment:
Where do we shut this down?

Reply:
Yes, this needs to be shutdown in stop method(), thanks for catching it.